src/rard/static/js/app-crit-editor.js

```diff
@@ -0,0 +1,271 @@
+          content: line.innerHTML,
+        };
+        fetch(g_update_apparatus_criticus_url, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            "X-CSRFToken": getCsrfToken(),
+          },
+          body: JSON.stringify(data),
+        })
+          .then((response) => {
+            // change button and disable editing
+            item.innerHTML = "Edit";
+            item.removeAttribute("data-editing");
+            line.setAttribute("contenteditable", false);
+            CKEDITOR.instances["appcrit" + id].destroy();
+            item.parentNode.querySelector(".cancelBtn").remove();
+            document
+              .querySelectorAll(".edit-app-crit-toggle")
+              .forEach(function (item) {
+                item.style.display = "block";
+              });
+          })
+          .catch((error) => console.error("Error:", error));
+      } else {
+        // Enable editing
+        line.setAttribute("contenteditable", true);
+        line.focus();
+        var original_content = line.innerHTML;
+        item.innerHTML = "Save";
+        item.setAttribute("data-editing", true);
+        document
+          .querySelectorAll(".edit-app-crit-toggle")
+          .forEach(function (item) {
+            item.style.display = "none";
+          });
+        item.style.display = "block";
+        var editor = CKEDITOR.inline(line, {
+          toolbar: [
+            {
+              name: "basicstyles",
+              items: [
+                "Bold",
+                "Italic",
+                "Underline",
+                "Superscript",
+                "Subscript",
+              ],
+            },
+          ],
+          setFocus: true,
+        });
+        // Add cancel button
+        cancelBtn = document.createElement("button");
+        cancelBtn.textContent = "Cancel";
+        cancelBtn.classList.add(
+          "cancelBtn",
+          "btn",
+          "btn-link",
+          "text-primary",
+          "btn-sm"
+        );
+        item.parentNode.insertBefore(
+          cancelBtn,
+          item.parentNode.querySelector(".delete-apparatus-criticus-line")
+        );
+        cancelBtn.addEventListener("click", () => {
+          line.setAttribute("contenteditable", false);
+          editor.destroy();
+          line.innerHTML = original_content;
+          item.innerHTML = "Edit";
+          item.removeAttribute("data-editing");
+          cancelBtn.remove();
+          document
+            .querySelectorAll(".edit-app-crit-toggle")
+            .forEach(function (item) {
+              item.style.display = "block";
+            });
+        });
+      }
+    });
+  });
+}
+
+function updateOriginalTextReferences() {
+  var originalTextEditor = CKEDITOR.instances["id_content"];
+  var content = originalTextEditor.getData();
+  var mentions = content.match(/data-id="\d+"/g);
+  var refSpans = content.match(/<span><span>#<\/span>\d+<\/span>/g);
```

Comment:
I get refSpans = null error when it gets to this function. Maybe the regex isn't quite right?

On the one I'm looking at, there are spaces around the number, so this might cover either case:
`/<span><span>#<\/span>\s*\d+\s*<\/span>/g`